src/main/kotlin/no/skatteetaten/aurora/boober/service/OpenShiftObjectGenerator.kt

```diff
@@ -96,8 +99,22 @@ class OpenShiftObjectGenerator(
 
     fun generateDeploymentConfig(auroraDeploymentSpec: AuroraDeploymentSpec,
                                  labels: Map<String, String>,
-                                 mounts: List<Mount>?): JsonNode? =
-            DeploymentConfigGenerator(mapper, velocityTemplateJsonService).create(auroraDeploymentSpec, labels, mounts)
+                                 mounts: List<Mount>?): JsonNode? {
+
+        if (auroraDeploymentSpec.deploy == null) {
+            return null
+        }
+        val deployment = auroraDeploymentSpec.applicationPlatform.handler.handleAuroraDeployment(auroraDeploymentSpec, labels, mounts)
```

Comment:
This was very well put and it is the kind of feedback that I love. <3

I think we should sit down and sketch this out in more detail. ADC has just grown dynamically as more stuff was needed and it could benefit from some refactoring.

Getting this abstraction right is very important. 

I will start with seperating the bounded contexts into their own packages. For now that is vault, auroraconfig, boober. Not sure if boober can be split up more.